libpromises/crypto.c

```diff
@@ -502,10 +502,29 @@ size_t CipherTextLengthMax(const EVP_CIPHER* cipher, size_t plaintext_length)
     return plaintext_length + (CipherBlockSizeBytes(cipher) * 2) - 1;
 }
 
+size_t PlainTextLengthMax(const EVP_CIPHER* cipher, size_t ciphertext_length)
+{
+    if(ciphertext_length <= 0)
+    {
+        ProgrammingError("PlainTextLengthMax: ciphertext_length is not positive (%ld)",
+                         ciphertext_length);
+    }
+
+    // check for potential integer overflow, leave some buffer
+    if(ciphertext_length > INT_MAX - 1024)
```

Comment:
Makes sense, as size_t is unsigned this check is useless. Removed.
Also changed the overflow check.